src/models/account.cpp

```diff
@@ -474,25 +477,29 @@ int Account::importFromCSV(const std::string& path)
             transaction.setAmount(amount);
             transaction.setGroupId(gid);
             addTransaction(transaction);
+
+            //Add Group if needed
+            if (getGroupById(gid) == std::nullopt && gid != -1) {
+                Group group{ (unsigned int) gid };
+                group.setName(groupName);
+                group.setDescription(groupDescription);
+                addGroup(group);
+            }
+
             imported++;
         }
     }
+    updateGroupAmounts();
     return imported;
 }
 
 void Account::updateGroupAmounts()
 {
-    for(std::pair<const unsigned int, Group>& pair : m_groups)
-    {
-        pair.second.setBalance(0);
-    }
-    for(const std::pair<const unsigned int, Transaction>& pair : m_transactions)
+    //Query for balance in SQL
+    SQLite::Statement qryGetGroupsBalance{ *m_db, "SELECT g.id, CAST(COALESCE(SUM(IIF(t.type=1, -t.amount, t.amount)), 0) AS TEXT) FROM transactions t RIGHT JOIN groups g on g.id = t.gid GROUP BY g.id;" };
+    while(qryGetGroupsBalance.executeStep())
     {
-        if(pair.second.getGroupId() != -1)
-        {
-            Group group{ m_groups.at(pair.second.getGroupId()) };
-            group.setBalance(group.getBalance() + (pair.second.getType() == TransactionType::Income ? pair.second.getAmount() : (pair.second.getAmount() * -1)));
-            updateGroup(group);
-        }
+        int index{ qryGetGroupsBalance.getColumn(0).getInt() };
+        m_groups[index].setBalance(boost::multiprecision::cpp_dec_float_50(qryGetGroupsBalance.getColumn(1).getString()));
```

Comment:
When dealing with `std::map` types please use `.at()` instead of `[]`
So 
`m_groups[index]` --> `m_groups.at(qryGetGroupsBalance.getColumn(0).getInt()).setBalance()...` 
(you can save the allocation of creating an `index` variable as well.

Reply:
For the allowation of the index, I feel like it is more readable, isn't it? The line would go be quite long if it was added in the `at()`

Reply:
It is more readable but we save an allocation of 4 byte :)) 
I've learned to write for performant code instead of readable working with C++ haha.

Reply:
You could also change the query variable name to just qryGroup to save some space.

Reply:
I was just saying, easier to maintain haha, especially for only 4 bytes